src/test/java/jraft/rpc/RpcServerTest.java

```diff
@@ -52,21 +61,46 @@ public void tearDown() {
     public void testAppendEntry() throws Exception {
         String leaderId = "LEADER_0";
         long term = 55;
+        when(raftServerContextImpl.getTerm()).thenReturn(term).thenReturn(term+1);
         RaftGrpc.RaftBlockingStub blockingStub = RaftGrpc.newBlockingStub(inProcessChannel);
         AppendEntryResponse response = blockingStub.append(AppendEntryRequest
                 .newBuilder().setLeaderId(leaderId).setTerm(term).build());
         assertEquals(term, response.getTerm());
         assertTrue(response.getSuccess());
+
+        response = blockingStub.append(AppendEntryRequest
+                .newBuilder().setLeaderId(leaderId).setTerm(term).build());
+        assertEquals(term+1, response.getTerm());
+        assertFalse(response.getSuccess());
     }
 
     @Test
     public void testVote() throws Exception {
         String candidateId = "candidate_0";
+        String candidateIdVoted = "candidate_1";
         long term = 55;
+        when(raftServerContextImpl.getTerm()).thenReturn(term).thenReturn(term).thenReturn(term+1).thenReturn(term+1);
+        when(raftServerContextImpl.getLastVoteFor())
+                .thenReturn(null).thenReturn(candidateIdVoted).thenReturn(null).thenReturn(candidateIdVoted);
         RaftGrpc.RaftBlockingStub blockingStub = RaftGrpc.newBlockingStub(inProcessChannel);
         VoteResponse response = blockingStub.requestVote(VoteRequest
                 .newBuilder().setCandidateId(candidateId).setTerm(term).build());
         assertEquals(term, response.getTerm());
         assertTrue(response.getVoteGranted());
+
```

Comment:
would be better if we add some short comments in the test to show what we're testing.